agent/xds/clusters.go

```diff
@@ -547,43 +552,70 @@ func makeClusterFromUserConfig(configJSON string) (*envoy.Cluster, error) {
 	return &c, err
 }
 
-func (s *Server) makeGatewayCluster(cfgSnap *proxycfg.ConfigSnapshot, clusterName string) (*envoy.Cluster, error) {
-	return s.makeGatewayClusterWithConnectTimeout(cfgSnap, clusterName, 0)
-}
-
-// makeGatewayClusterWithConnectTimeout initializes a gateway cluster
-// with the specified connect timeout. If the timeout is 0, the connect timeout
-// defaults to use the configured gateway timeout.
-func (s *Server) makeGatewayClusterWithConnectTimeout(cfgSnap *proxycfg.ConfigSnapshot,
-	clusterName string, connectTimeout time.Duration) (*envoy.Cluster, error) {
+func (s *Server) makeGatewayCluster(snap *proxycfg.ConfigSnapshot,
+	name string, hostnameEndpoints structs.CheckServiceNodes, isRemote, onlyPassing bool, connectTimeout time.Duration) *envoy.Cluster {
```

Comment:
This function has a bunch of args now. It _might_ be worth it for clarity to use an options struct, especially since there are now two boolean args which are easy to mix up.